typing/types.mli

```diff
@@ -264,7 +264,7 @@ and value_kind =
 
 and value_unbound_reason =
   | Val_unbound_instance_variable
-  | Val_unbound_ghost_recursive
+  | Val_unbound_ghost_recursive of bool
```

Comment:
If we keep this aspect of the patch, I would prefer to see an interface that more explicitly gives a meaning to the `bool` payload, for example using a record type (possibly an inline record) with a field name. Currently it is very unclear to a reader of the `.mli` what this `bool` would mean.

Reply:
Sure; I wanted to push this draft to get some feedback
and explore the design space. For instance @lpw25 
suggested that we could use the recursive check to ensure
that the hint actually makes sense.